.githooks/pre-commit

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env sh
+RED="\033[1;31m"
+BLUE="\033[34m"
+GREEN="\033[1;32m"
+
+linter_exit_code=0
+eslint_files=$(git diff --cached --diff-filter=d --name-only --line-prefix=`git rev-parse --show-toplevel`/ | grep '.js$\|.ts$\|.vue$')
+stylelint_files=$(git diff --cached --diff-filter=d --name-only --line-prefix=`git rev-parse --show-toplevel`/ | grep '.css$\|.vue$')
+
+echo "${BLUE}(pre-commit) Running linters..."
+if [ "$eslint_files" ]
```

Comment:
Running lint pre commit might need the typescript files need to be built first. (there are some eslint checks for imports)

I don't think it would be as much an issue once we've migrated as much as possible to typescript though.

Should we add `npm run clean` and `npm run build:backend` on every commit as well for now? Seems like a huge hindrance to the development experience but also a temporary one.

Reply:
@ang-zeyu 
Personally, I think that would be fine.  Experienced developers that want to skip the check can always use the no verify flag.

Reply:
Also a bunch of git UIs that don't have this option unfortunately. 😢

Reply:
(let's add it then)